src/device.cpp

```diff
@@ -73,10 +73,9 @@ void cvk_device::init_vulkan_properties(VkInstance instance) {
         auto corever = std::get<0>(ver_ext_prop);
         auto ext = std::get<1>(ver_ext_prop);
         auto prop = std::get<2>(ver_ext_prop);
-        if ((corever != 0) && (m_properties.apiVersion >= corever)) {
-            prop->pNext = pNext;
-            pNext = prop;
-        } else if ((ext != nullptr) && is_vulkan_extension_enabled(ext)) {
+        if (((ext != nullptr) && is_vulkan_extension_enabled(ext)) ||
+            ((ext == nullptr) && (corever != 0) &&
+             (m_properties.apiVersion >= corever))) {
```

Comment:
This doesn't look right to me. It means we're only using core features for which clvk has no knowledge of a corresponding extension. I don't think implementations are required to report extensions that were promoted (worth double-checking though).

Reply:
I understand.
It is an improvement over what we had before https://github.com/kpet/clvk/commit/a55be50a0e8f5a2ea855d8832c7989cb385f246b. But it is definitely not what we would prefer (which is the actual version).
If we are to keep it that way, I don't know what I will do for google infra. I don't want to disable all swiftshader tests, but I am not sure if anything else will be possible to do easily.

Reply:
Just a thought: would it be possible to actually implement those features in swiftshader?

Reply:
Good point, I'll look at it

Reply:
One of those has already been implemented upstream, we just need to rebuild swiftshader.
The other is `PhysicalDevice8BitStorageFeatures`.

Reply:
I'll kick off an update of swiftshader.

Reply:
> Functionality in this extension is included in core Vulkan 1.2, with the KHR suffix omitted. However, if Vulkan 1.2 is supported and this extension is not, the StorageBuffer8BitAccess capability is optional. The original type, enum and command names are still available as aliases of the core functionality.
https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VK_KHR_8bit_storage.html#_promotion_to_vulkan_1_2

Reply:
So I was right and reporting the extension is not required (at least for that one but that's a good hint that there's no blanket requirement). We can't make the change proposed in this PR then.

Reply:
If I understand correctly, our way forward is to add support for `VkPhysicalDevice8BitStorageFeatures` in swiftshader. But it will just report that nothing is supported, right?

Reply:
Yes, that's the cheapest possible change but maybe swiftshader actually supports those features? You'd have to check (with them). BTW, I've now updated switshader in the CI, that should remove one of the failures.

Reply:
Update:
I was mistaken, nothing has been added upstream. I was only seeing one of them in google infra because of the stop at first error.
The other one is `PhysicalDeviceShaderFloat16Int8Features`.

Both of them are not supported in swiftshader. But I feel like they are just missing from one switch-case.
I am testing it, and will propose a change in swiftshader.

Reply:
Cool, thanks for the update!